File: Easy/94_binary_tree_inorder_traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), 
left(left), right(right) {}
 * };
 */
class Solution {
public:
    void getVal(TreeNode* tree, vector<int>* vec) {
        if(tree == NULL) return;
        getVal(tree->left, vec);
        vec->push_back(tree->val);
        getVal(tree->right, vec);
    }
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> inorder;
        getVal(root, &inorder);
        return inorder;
    }
};

//inorder => Left->Data->Right

File: python/easy/94_binary_tree_inorder_traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorder(self, root: Optional[TreeNode], arr: List[int]):
        if root == None: return
        self.inorder(root.left, arr)
        arr.append(root.val)
        self.inorder(root.right, arr)

    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        answer = []
        self.inorder(root, answer)
        return answer
